lib/Analysis/GateAnalysis.cc

```diff
@@ -230,15 +223,32 @@ void GateAnalysisImpl::processPhi(PHINode *PN, Instruction *insertionPt) {
 
       // Or the successors unconditionally flows to an already processed block.
       // Note that there can be at most one such block.
-      for (auto &BlockValuePair : flowingValues) {
-        if (m_PDT.dominates(BlockValuePair.first, S)) {
-          SuccToVal[S] = BlockValuePair.second;
-          GSA_LOG(errs() << "2) SuccToVal[" << S->getName()
-                         << "] = postdom for cd "
-                         << BlockValuePair.first->getName() << ": "
-                         << SuccToVal[S]->getName() << "\n");
+      bool found = false;
+      BasicBlock *bb =  S;
+      auto *node = m_PDT.getNode(bb)->getIDom();
+      while (node != nullptr) {
+        auto it = flowingValues.find(bb);
+        if (it != flowingValues.end()) {
```

Comment:
nit: `if (flowingValues.count(bb) == 0)`

Reply:
and also in the code below

Reply:
I don't understand the benefit of this. This will require two accesses to `flowingValues` in case the value is already present: first to check if the value is present and second to get the value.

Reply:
right, I missed the use below